lib/utils/qemu.coffee

```diff
@@ -61,26 +61,40 @@ getQemuPath = ->
 		Promise.resolve(fs.mkdir(binDir))
 		.catch(code: 'EEXIST', ->)
 		.then ->
-			path.join(binDir, QEMU_BIN_NAME)
+			path.join(binDir, "#{QEMU_BIN_NAME}-#{arch}")
 
 platformNeedsQemu = ->
 	os = require('os')
 	os.platform() == 'linux'
 
-installQemu = ->
+installQemu = (arch) ->
 	request = require('request')
 	fs = require('fs')
 	zlib = require('zlib')
+	tar = require('tar-stream')
 
-	getQemuPath()
+	getQemuPath(arch)
 	.then (qemuPath) ->
 		new Promise (resolve, reject) ->
 			installStream = fs.createWriteStream(qemuPath)
-			qemuUrl = "https://github.com/balena-io/qemu/releases/download/#{QEMU_VERSION}/#{QEMU_BIN_NAME}.gz"
+			downloadArchiveName = "qemu-3.0.0+resin-#{arch}.tar.gz"
+			qemuUrl = "https://github.com/balena-io/qemu/releases/download/#{QEMU_VERSION}/#{downloadArchiveName}"
+			extract = tar.extract()
+			extract.on('entry', (header, stream, next) ->
```

Comment:
you can drop the parens on this call if you like

Reply:
Ah, coffeescript. Thanks!